### PR TITLE
[CR]Remove hunger in favor of just calories

### PR DESCRIPTION
**Pull request by Coolthulhu**
Remove the `get/set/mod`+`_hunger` functions in `Character`.
They were just wrappers around kcal anyway.

I'm not sure if I didn't scramble any of the replacement values (signs, math), hence [CR].
For values where it doesn't matter, I used something from 5 to 10 (kcal per hunger).
For ones where it does, I used 10.

The actual value for closest conversion is ~8.6. This corresponds to 2500 kcal per day.

I'm not sure if I should change effects here or leave them with conversions for now.

### CODE REVIEW

**Comment by SzQ1** [comment]
So what actually changes from the player perspective? Less food needed?

**Comment by Coolthulhu** [comment]
No real change from player's perspective, at most ~20% change for hunger/satiation induced by effects that cause it.

**Comment by SzQ1** [comment]
So it's for simplification of the code without much of altering gameplay wise, basically?

File: src/npctalk.cpp [hunk 1]
```diff
@@ -1096,8 +1096,10 @@ std::string dialogue::dynamic_line( const talk_topic &the_topic ) const
             } else {
                 info += _( "\nThirsty" );
             }
-            if( p->get_hunger() < 100 ) {
-                time_duration hunger_at = 5_minutes * ( 100 - p->get_hunger() ) / rates.hunger;
+            if( p->max_stored_calories() - p->get_stored_kcal() > 500 ) {
+                time_duration hunger_at = 5_minutes
+                                          * ( 500 - p->max_stored_calories() + p->get_stored_kcal() )
+                                          / rates.kcal;
```
Comment (olanti-p) [inline src/npctalk.cpp:1102]: Is `rates.kcal` ever calculated?
The only place I see it be used is assignment of placeholder value
https://github.com/cataclysmbnteam/Cataclysm-BN/blob/33a41a8f69e4199445dc9fc58b44fb1c43ad65b6/src/character.cpp#L4745-L4746

File: tests/new_character_test.cpp [hunk 2]
```diff
@@ -71,7 +71,7 @@ static avatar get_sanitized_player()
     ret.recalc_hp();
 
     // Set these insanely high so can_eat doesn't return TOO_FULL
-    ret.set_hunger( 10000 );
+    ret.set_stored_kcal( 100 );
     ret.set_thirst( 10000 );
```
Comment (olanti-p) [inline tests/new_character_test.cpp:75]: Nitpick: comment doesn't make sense anymore

File: src/npctalk.cpp (hunk 1 above)
Comment (Coolthulhu) [reply src/npctalk.cpp:1102]: Yep, unused. I'll change the line and remove the field.

**Pull request merged**